### PR TITLE
Fix incorrectly labeled music tracks

### LINKED ISSUE #221: Fix the name of music tracks
There's a list that [matches music tracks ids to names](https://github.com/zladx/LADX-Disassembly/blob/master/src/constants/sfx.asm#L5) – but it seems to be invalid: many names are not the correct ones. (See [src/constants/sfx.asm](https://github.com/zladx/LADX-Disassembly/blob/master/src/constants/sfx.asm#L5))

By listening to each music track, the constants could be renamed to have the correct names.

## How

Listening to a specific music track id in-game is easy:

1. Run the game in an emulator with a debugger (like BGB);
2. Go to the overworld;
3. Edit the memory at `$FFB0` (`hMusicTrack`), and write a track ID;
4. The music track should start playing.

### PR DESCRIPTION
**Pull request by Vextrove**
This fixes a good portion of the mislabeled tracks.
The commented tracks are still under investigation. The notes are to help me quickly find themes as I try to to match them. I will do a second PR when I verify all or most of the tracks.   

Fix #221

### CODE REVIEW

File: src/sfx.asm [hunk 1]
```diff
@@ -0,0 +1,192 @@
+;
+; Constants for audio effects
+
+; Values for hMusicTrack
+ 
+MUSIC_NONE                                      equ $00
```
Comment (kemenaran) [inline src/sfx.asm:6]: It seems all those constants below are duplicated from the ones declared in `constants/sfx.asm`.

I'd rather have the constants only in a single place. Is it possible for you to delete those duplicate declarations in `constants/sfx.asm`?

File: src/constants/sfx.asm [hunk 2]
```diff
@@ -1,102 +1,109 @@
...
+MUSIC_TURTLE_ROCK                               equ $5A
+MUSIC_EAGLES_TOWER                              equ $5B
+MUSIC_BEFORE_EAGLES_TOWER_BOSS                  equ $5C;    Plays before the boss at the top of Eagle's Tower is summon by Grim Creeper.
+MUSIC_FINAL_BOSS_DIALOG                         equ $5D;	Plays during the final boss' dialog.
+MUSIC_BOSS_WARNING                              equ $5E;	Plays when the bosses tell you not to wake the windfish after defeating them.
+MUSIC_FINAL_BOSS_DEFEATED                       equ $5F
+MUSIC_ZELDA_NICKNAME_EASTER_EGG                 equ $60;	ZELDA nickname easter egg.
+MUSIC_COLOR_DUNGEON                             equ $61;
+;MUSIC_COLOR_DUNGEON_COPIES                     equ $62 to 6F;	Same as $61	
+;MUSIC_GLITCHED                                 equ $70 to $FE;	Glitches the music that played before it.
```
Comment (kemenaran) [inline src/constants/sfx.asm:105]: This commented statement looks like a good way of signaling glitched music tracks 👍

File: src/constants/sfx.asm [hunk 3]
```diff
@@ -1,102 +1,109 @@
...
+MUSIC_DREAM_SHRINE                              equ $53
+;MUSIC_UNKNOWN                                  equ $54; intense arpeggio, pre boss music? Switches to $19.
+MUSIC_FLYING_ROOSTER_REVIVED                    equ $55
+;MUSIC_UNKNOWN                                  equ $56; punch noises in upwards chord progression, plays $55 afterwards
+;MUSIC_CUCCO_HOUSE                              equ $57;	Is called 'Cucco House'
+MUSIC_FACE_SHRINE                               equ $58
+;MUSIC_WINDFISH                                 equ $59;	Is called 'The WindFish'
+MUSIC_TURTLE_ROCK                               equ $5A
+MUSIC_EAGLES_TOWER                              equ $5B
+MUSIC_BEFORE_EAGLES_TOWER_BOSS                  equ $5C;    Plays before the boss at the top of Eagle's Tower is summon by Grim Creeper.
```
Comment (kemenaran) [inline src/constants/sfx.asm:98]: As a nice-to-have, if you still feel like polishing this:

For consistency, could you ensure that all comments follow the same format? Something like `MUSIC_CONSTANT     equ $VALUE ; Capitalized comment` (with a space before and after the semicolon)?

(But if you feel like you already worked much on this, I can merge it as-this.)

File: src/sfx.asm (hunk 1 above)
Comment (Vextrove) [reply src/sfx.asm:6]: I think I accidentally upladed it to the wrong folder, let me check.

File: src/constants/sfx.asm (hunk 3 above)
Comment (Vextrove) [reply src/constants/sfx.asm:98]: I used a tab I was sure of my comment.
The lowercase ones are notes to help me quickly find tracks when I discover where they are used in the game. That's why some of them have/had oddly specific descriptions; if a particular sound evoked a really specific feeling I would be able to quickly tell where that track is in the list by looking for the note that describes the same feeling.
Once I'm done figuring all of them out, I will format it correctly.

File: src/sfx.asm (hunk 1 above)
Comment (Vextrove) [reply src/sfx.asm]: constants/sfx.asm has been removed.

File: src/constants/sfx.asm [hunk 4]
```diff
@@ -1,102 +1,109 @@
 ;
 ; Constants for audio effects
 
-; @TODO Are these correct? Some of these don't seem to match
-; their actual in-game use (???)
-
-; Values for hMusicTrack
+; Values for wActiveMusicTrack
```
Comment (kemenaran) [inline src/constants/sfx.asm:4]: 👍

**Comment by kemenaran** [comment]
Looks good to me. Thanks!

**Pull request merged**